### ISSUE TITLE
False positive in `useless_let_if_seq`

### ISSUE THREAD

**Issue by dtolnay** [Manishearth/rust-clippy#985]
#979 fixed a false positive when the variable is used in the condition of the if-statement: [let_if_seq.rs#L74](https://github.com/Manishearth/rust-clippy/pull/979/files#diff-b763fc6873a33e513891a686cffb0c72R74). There remains a false positive when the variable is used in the then- or else-clause but not the condition:

```rust
#![feature(plugin)]
#![plugin(clippy)]

fn main() {
    let mut x = f();
    if cond() {
        x = g(x);
    }
    println!("{}", x);
}

fn f() -> i32 { 0 }
fn g(x: i32) -> i32 { x + 1 }
fn cond() -> bool { true }
```

The suggested replacement is `let <mut> x = if cond() { g(x) } else { f() };` which is not correct.
